src/core/src/pass/graph_rewrite.cpp

```diff
@@ -151,7 +152,16 @@ bool ov::pass::GraphRewrite::apply_matcher_passes(std::shared_ptr<Model> f,
 
         // Apply MatcherPass. In case if it returns true no other MatcherPasses will apply
         // to this node
-        bool status = m_pass->apply(node);
+
+        bool status = false;
+        {
+            if (m_pass->skip_profiling()) {
+                status = m_pass->apply(node.lock());
+            } else {
+                PassTracker tracker(f, m_pass->get_name());
+                status = m_pass->apply(node.lock());
```

Comment:
I would suggest to move real profiling code under some preprocessor macro to be able to cut such code completely off to avoid any binary size increase doubts. It should be controlled by a new pre-processing frag. I think this particular place in the code is a starting point where we should apply this conditional compilation and include all the code solely dedicated to such profiling.

I think that a customer of this service is a transformation developer who really needs to rebuild openvino, so providing a cmake flag and rebuild is not a big deal.